registry/client.go

```diff
@@ -352,3 +352,118 @@ func (e Error) Error() string {
 
 	return "registry error: " + strconv.Itoa(e.StatusCode)
 }
+
+// Compatibility levels.
+const (
+	BackwardCL           string = "BACKWARD"
+	BackwardTransitiveCL string = "BACKWARD_TRANSITIVE"
+	ForwardCL            string = "FORWARD"
+	ForwardTransitiveCL  string = "FORWARD_TRANSITIVE"
+	FullCL               string = "FULL"
+	FullTransitiveCL     string = "FULL_TRANSITIVE"
+	NoneCL               string = "NONE"
+)
+
+/*
+this function returns whether or not a compatibility level
+is valid according to the ones described in:
+https://docs.confluent.io/platform/current/schema-registry/develop/api.html#compatibility
+.
+*/
+func validCompatibilityLevel(compatibilityLevel string) bool {
+	switch compatibilityLevel {
+	case BackwardCL:
+		return true
+	case BackwardTransitiveCL:
+		return true
+	case ForwardCL:
+		return true
+	case ForwardTransitiveCL:
+		return true
+	case FullCL:
+		return true
+	case FullTransitiveCL:
+		return true
+	case NoneCL:
+		return true
+	default:
+		return false
+	}
+}
+
+// compatibilityPayload is the response body obtained from the schema registry when updating a compatibility level.
+type compatibilityPayload struct {
+	Compatibility string `json:"compatibility"`
+}
+
+/*
+errByInvalidCL returns error for and invalid compatibility level, i.e. a compatibility
+level not present in:
+https://docs.confluent.io/platform/current/schema-registry/develop/api.html#compatibility .
+*/
+func errByInvalidCL(compatibilityLevel string) error {
+	return fmt.Errorf("invalid compatibility level %s", compatibilityLevel)
+}
+
+// PutCompatibilityLevelGlobal sets the global compatibility level of the registry.
+func (c *Client) PutCompatibilityLevelGlobal(
+	ctx context.Context,
+	compatibilityLevel string,
+) (string, error) {
+	if !validCompatibilityLevel(compatibilityLevel) {
+		return "", errByInvalidCL(compatibilityLevel)
+	}
+	var payload compatibilityPayload
+	inPayload := compatibilityPayload{Compatibility: compatibilityLevel}
+	err := c.request(ctx, http.MethodPut, "config/", inPayload, &payload)
+	if err != nil {
+		return "", err
+	}
+
+	return payload.Compatibility, nil
+}
+
+// PutCompatibilityLevel sets the compatibility level of a subject.
+func (c *Client) PutCompatibilityLevel(
+	ctx context.Context,
+	subject, compatibilityLevel string,
+) (string, error) {
+	if !validCompatibilityLevel(compatibilityLevel) {
+		return "", errByInvalidCL(compatibilityLevel)
+	}
+	var payload compatibilityPayload
+	inPayload := compatibilityPayload{Compatibility: compatibilityLevel}
+	err := c.request(ctx, http.MethodPut, "config/"+subject, inPayload, &payload)
+	if err != nil {
+		return "", err
+	}
+
+	return payload.Compatibility, nil
+}
+
+// GetCompatibilityLevelGlobal gets the global compatibility level.
+func (c *Client) GetCompatibilityLevelGlobal(
```

Comment:
`GetGlobalCompatibilityLevel` would be better.

Reply:
done